spec/globus_client_spec.rb

```diff
@@ -119,12 +119,17 @@
       before do
         allow(described_class::Identity).to receive(:new).and_return(fake_identity)
         allow(GlobusClient::Authenticator).to receive(:token).and_return("a_token", "new_token")
-        allow(fake_identity).to receive(:exists?).once.and_raise(GlobusClient::UnexpectedResponse::UnauthorizedError)
-        allow(fake_identity).to receive(:exists?).once.and_return(true)
+        response_values = [:raise, true]
+        allow(fake_identity).to receive(:exists?).twice do
+          v = response_values.shift
+          (v == :raise) ? raise(GlobusClient::UnexpectedResponse::UnauthorizedError) : v
+        end
       end
```

Comment:
this ensures it calls the raise first, and then returns true

Reply:
Nice! This is more or less what's in the RSpec docs, I think: https://relishapp.com/rspec/rspec-mocks/docs/configuring-responses/block-implementation#simulating-a-transient-network-failure